src/mono/wasm/debugger/BrowserDebugProxy/MonoSDBHelper.cs

```diff
@@ -2211,16 +2266,28 @@ public async Task<JArray> GetArrayValues(SessionId sessionId, int arrayId, Cance
             var commandParamsWriter = new MonoBinaryWriter(commandParams);
             commandParamsWriter.Write(arrayId);
             commandParamsWriter.Write(0);
-            commandParamsWriter.Write(length);
+            commandParamsWriter.Write(length.TotalLength);
             var retDebuggerCmdReader = await SendDebuggerAgentCommand<CmdArray>(sessionId, CmdArray.GetValues, commandParams, token);
             JArray array = new JArray();
-            for (int i = 0 ; i < length ; i++)
+            for (int i = 0 ; i < length.TotalLength; i++)
             {
-                var var_json = await CreateJObjectForVariableValue(sessionId, retDebuggerCmdReader, i.ToString(), false, -1, false, token);
+                var var_json = await CreateJObjectForVariableValue(sessionId, retDebuggerCmdReader, length.GetArrayIndexString(i), false, -1, false, token);
                 array.Add(var_json);
             }
             return array;
         }
+
+        public async Task<JObject> GetArrayValuesProxy(SessionId sessionId, int arrayId, CancellationToken token)
+        {
+            var length = await GetArrayLength(sessionId, arrayId, token);
+            var arrayProxy = JObject.FromObject(new
+            {
+                items = await GetArrayValues(sessionId, arrayId, token),
```

Comment:
`GetArrayValues` also calls `GetArrayLength`. Instead maybe `GetArrayValues` can return `(JArray, ArrayDimensions)`, or it can take a `ArrayDimensions argument?